### PR TITLE
[Hub Generated] Review request for Microsoft.ManagedServices to add version preview/2022-01-01-preview

### PR DESCRIPTION
**Pull request by skayani**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/skayani/azure-rest-api-specs/managedservicesqueryapi...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by skayani** [comment]
@akning-ms Not sure how to resolve the Python SDK issues. Can you give some pointers?

**Comment by akning-ms** [comment]
> @akning-ms Not sure how to resolve the Python SDK issues. Can you give some pointers?

You are saying pythons SDK breaking change? @msyyc to help python SDK issue. 
BTW, you may also need to fix linter and prettier check issue and looks you renamed "registrationDefinitionIdParameter" which may need a breaking change review board review

**Comment by msyyc** [comment]
approved for python breaking change

**Comment by skayani** [comment]
Hi @akning-ms , can you please ack on this PR? Once you do, I will update our RP manifest with the new version so that we can then merge this

**Comment by akning-ms** [comment]
> e sure this PR is merge-ready (but don't merge yet)? We will give ack on merge as soon as the new API is live.

Hi @skayani, For registrationDefinitionIdParameter, as it is breaking your latest stable version, if your previous preview version is not get breaking change review board approval, you may have to get breaking change review board approval. if yes. can you   show me the PR# for previous preivew version's approval?

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json
Comment (akning-ms) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:1241]: For SystemData, recommend to ref to common-type definition if possible：
Ref to specification\common-types\resource-management\v3\types.json

Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:1241]: > > e sure this PR is merge-ready (but don't merge yet)? We will give ack on merge as soon as the new API is live.
> 
> Hi @skayani, For registrationDefinitionIdParameter, as it is breaking your latest stable version, if your previous preview version is not get breaking change review board approval, you may have to get breaking change review board approval. if yes. can you show me the PR# for previous preivew version's approval? And for systemData, it is ok now, but do you have a roughly plan to support it?

HI @akning-ms , [here's the PR](https://github.com/Azure/azure-rest-api-specs/pull/8913) where we introduced the 2020-02-01-preview API. We didn't need breaking changes review. For the SystemData question, there aren't plans to support it now as the MarketplaceRegistrationDefinition resource isn't customer-accessible (it's an internally used resource). We may revisit this in the near future.

Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:1241]: Also @akning-ms , what exactly do I use for the $ref path value? I couldn't find any other RPs using that path

Comment (akning-ms) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:1241]: Thanks Skayani, as the PR for 2020-02-01-preivew is merged before our cross API breaking change((mainly track the break for stable version)
) enable. Looks there is no any approval from breaking change review board.  so you may need to get breaking change review board approval, you can follow guide to do it with some justification andcurrent status of 2020-02-01-preview. e.g is there any SDK released with the version> . https://github.com/Azure/azure-rest-api-specs/pull/17089#issuecomment-1018250236

Comment (akning-ms) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:1241]: For systemData, you can refer to this example:
https://github.com/Azure/azure-rest-api-specs/blob/e88a7008354fa35cdb3c551d6c7f9eb6b1e21db0/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2021-09-01/extensions.json#L691
systemData is same for v2 v3, just ensure use same version among your swagger

Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json]: Hey @akning-ms , we would rather avoid the breaking change review so I have reverted the offending breaking change. Please let me know when the PR is merge ready as I have made your suggestions

Comment (akning-ms) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json]: Thanks skayani, After revert, breaking change review is not required, Once ARM sign off the PR, I will merge this PR.
If urgent:  please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json
Comment (mentat9) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json:656]: >v2 [](http://example.com/codeflow?start=67&length=2)

Any reason you aren't using v3 for this?

**Comment by mentat9** [comment]
Signed off with comments from ARM perspective.

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json]: No reason, just reused that reference from an example. Changed it to v3, thanks

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json
Comment (skayani) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2022-01-01-preview/managedservices.json]: @akning-ms Now that we have ARM review, are we good to go?

**Review by akning-ms — APPROVED** [review]

**Comment by akning-ms** [comment]
Hi @skayani, I have approved your PR.  Seems your service is not ready. Can I know when it will be deployed?  I will help merge it when your service is deploy to production.

**Comment by skayani** [comment]
> Hi @skayani, I have approved your PR. Seems your service is not ready. Can I know when it will be deployed? I will help merge it when your service is deploy to production.

Hey @akning-ms , I kicked off the RP manifest update job last week. I see the changes are in prod now, can we merge?

**Comment by skayani** [comment]
@akning-ms Ping again, can we please merge the PR?

**Pull request merged**
